addons/l10n_es_edi_facturae/models/account_edi_format.py

```diff
@@ -0,0 +1,499 @@
+        return items, taxes, totals
+
+    @api.model
+    def l10n_es_edi_facturae_export_facturae(self, invoice):
+        """
+        Produce the Facturae XML data for the invoice.
+
+        :param account.move invoice: The invoice to export.
+        :return tuple: (data needed to render the full template, data needed to render the signature template).
+        """
+
+        self_company = invoice.company_id
+        partner = invoice.partner_id.parent_id or invoice.partner_id
+
+        if not self_company.l10n_es_edi_facturae_tax_identifier:
+            raise UserError(_('The company needs a set tax identification number or VAT number'))
+
+        if not partner.l10n_es_edi_facturae_tax_identifier:
+            raise UserError(_('The partner needs a set tax identification number or VAT number'))
+
+        def fmt_eur(value):
+            # Format the monetary values to avoid trailing decimals (e.g. 90.850...1).
+            return float_repr(value, eur_curr.decimal_places)
+
+        def fmt_curr(number):
+            # Format the monetary values to avoid trailing decimals (e.g. 90.850...1).
+            if inv_curr.name == 'EUR':
+                return fmt_eur(number)
+            return float_repr(number, inv_curr.decimal_places)
+
+        eur_curr = self.env['res.currency'].search([('name', '=', 'EUR')])
+        inv_curr = invoice.currency_id or self_company.currency_id
+        IndividualName = namedtuple('IndividualName', ['firstname', 'surname', 'surname2'])
+        firstname = surname = surname2 = ''
+        phone_clean_table = str.maketrans({" ": None, "-": None, "(": None, ")": None, "+": None})
+        legal_literals = invoice.narration.striptags() if invoice.narration else False
+        legal_literals = legal_literals.split(";") if legal_literals else False
+
+        if partner.is_company:
+            partner_name = IndividualName('', '', '')
+        else:
+            name_split = [part for part in partner.name.replace(', ', ' ').split(' ') if part]
+            if len(name_split) > 2:
+                firstname = ' '.join(name_split[:-2])
+                surname, surname2 = name_split[-2:]
+            elif len(name_split) == 2:
+                firstname = ' '.join(name_split[:-1])
+                surname = name_split[-1]
+
+            partner_name = IndividualName(firstname, surname, surname2)
+
+        template_values = EDI_XML_DICT_TEMPLATES['facturae'].copy()
+        self_party_values = EDI_XML_DICT_TEMPLATES['business'].copy()
+        partner_party_values = EDI_XML_DICT_TEMPLATES['business'].copy()
+        signature_values = EDI_XML_DICT_TEMPLATES['signature'].copy()
+
+        if invoice.move_type == "entry":
+            return False
+        invoice_issuer_type = 'EM' if invoice.move_type == 'out_invoice' else 'RE'
+        invoice_issuer_signature_type = 'supplier' if invoice.move_type == 'out_invoice' else 'customer'
+
+        company_is_not_euro = self_company.currency_id != eur_curr
```

Comment:
Should we really manage that? I think this EDI should be applied only on EUR companies instead